File: ui/src/importdeck/SaveUnregisteredDeck.ts
import { Expansion } from "../generated-src/Expansion"

export interface SaveUnregisteredDeck {
    name: string
    cards: CardsInHouses
    expansion: Expansion
}

export interface CardsInHouses {
    /**
     * House to card name
     */
    [key: string]: string[]
}

File: src/main/kotlin/coraythan/keyswap/decks/models/SaveUnregisteredDeck.kt
package coraythan.keyswap.decks.models

import coraythan.keyswap.House
import coraythan.keyswap.expansions.Expansion

data class SaveUnregisteredDeck(
        val cards: Map<House, List<String>>,
        val name: String,
        val expansion: Expansion = Expansion.CALL_OF_THE_ARCHONS
)
